src/ui/osx/TogglDesktop/Feature/GoogleLogin/GoogleAuthenticationServer.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  GoogleAuthenticationServer.swift
+//  TogglDesktop
+//
+//  Created by Nghia Tran on 9/17/19.
+//  Copyright © 2019 Alari. All rights reserved.
+//
+
+import Foundation
+import AppAuth
+import GTMAppAuth
+
+final class GoogleAuthenticationServer {
+
+    static let shared = GoogleAuthenticationServer()
+
+    private struct Constants {
+        static let Issuer = "https://accounts.google.com"
+        static let ClientID = "426090949585-uj7lka2mtanjgd7j9i6c4ik091rcv6n5.apps.googleusercontent.com"
+        static let ClientSecret = "6IHWKIfTAMF7cPJsBvoGxYui"
+        static let RedirectURI = "com.googleusercontent.apps.426090949585-uj7lka2mtanjgd7j9i6c4ik091rcv6n5:/oauthredirect"
+        static let TogglAuthorizerKey = "toggldesktop-authorization"
+        static let Scopes = [OIDScopeOpenID, OIDScopeProfile, OIDScopeEmail]
```

Comment:
Even though we don't receive the Email and Profile from users, but we have to use all scopes in order to successfully login to Toggl service.

Otherwise, it won't work.

Reply:
FYI in the Windows app we use 2 scopes: Email and Profile
https://github.com/toggl/toggldesktop/blob/c823ceacef0a84f3e705a562518e6b350c9bd79a/src/ui/windows/TogglDesktop/TogglDesktop/ui/views/LoginView.xaml.cs#L388-L396

Reply:
I tried to run it with 2 scopes and it worked fine, so you can remove `OIDScopeOpenID`

Reply:
Let me check again because according to their Documentation, it's required to ask the Google return OpenID.

Reply:
It's required only for OpenID Connect requests. I think it's not necessary for plain OAuth2, but I guess it won't hurt either.